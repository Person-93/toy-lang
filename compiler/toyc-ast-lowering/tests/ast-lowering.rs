use test_util::{assert_debug_snapshot, snapshots, TestSession};
use toyc_ast_lowering::HirContext;

fn main() {
  snapshots(|text| {
    let session = TestSession::new("test-ast-lowering");
    let ast = toyc_ast::parse_single_file(text, &session.handler).unwrap();
    let ctx = HirContext::new(&session);
    assert_debug_snapshot!(ctx.lower_package(&ast));
  });
}
