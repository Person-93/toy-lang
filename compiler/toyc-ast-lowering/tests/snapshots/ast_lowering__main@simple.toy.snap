---
source: compiler/toyc-ast-lowering/tests/ast-lowering.rs
assertion_line: 13
expression: ctx.root()

---
Package {
    items: [
        Function(
            Function {
                id: HirId(
                    1,
                    PhantomData,
                    PhantomData,
                ),
                ident: f @ 3..4,
                visibility: Inherited,
                self_kind: None,
                fn_type: FnType {
                    extern_: No,
                    unsafety: No,
                    constness: No,
                    generics: None,
                    inputs: [],
                    output: Default,
                },
                body_id: Some(
                    BodyId(
                        HirId(
                            2,
                            PhantomData,
                            PhantomData,
                        ),
                    ),
                ),
                span: Span {
                    lo: 0,
                    hi: 10,
                },
            },
        ),
    ],
    span: Span {
        lo: 0,
        hi: 10,
    },
}
